### PR TITLE
Migrate build chain

### LINKED ISSUE #840: 📋 [TASK] - Migrate Build Chain in Core AaC Repo
### Parent Feature

#838

### Task Description

As an AaC Developer, I want to update the core AaC repository to use the new build chain process, so that I have a consistent build chain across project repositories.

### Acceptance Criteria

- [ ] Replace setup.py file with a pyproject.toml file in core AaC repo.
- [ ] Edit other build and infrastructure files as needed.
  - [ ] Manifest.in
  - [ ] tox.ini
  - [ ] README
- [ ] Pull latest build workflows from organization templates as needed

### Unresolved questions

_No response_

### PR DESCRIPTION
**Pull request by bunchw**
# Description

Updating AaC to use the new build chain process.

# Linked Items:

Closes/Fixes/Resolves #840 

### Added

Added a pyproject.toml file, and a MANIFEST.in file

### Changed

- _Describe any changes in existing functionality._

### Deprecated

- _Describe any deprecated features._

### Removed

- _Describe any removed features._

### Fixed

- _Describe any bug fixes._

### Security

- _Describe any security-related changes._

# Checklist:

- [ ] I updated project documentation to reflect my changes.
- [ ] My changes generate no new warnings.
- [ ] I updated new and existing unit tests to account for my changes.
- [ ] I linked the associated item(s) to be closed.
- [ ] I bumped the version.
- [ ] I added the labels corresponding to my changes.

### CODE REVIEW

File: python/src/aac/__init__.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 import logging
 import os
 
-__version__ = "0.4.22"
+__version__ = "0.4.23"
```
Comment (lizzcondrey) [inline python/src/aac/__init__.py]:
```suggestion
__version__ = "0.4.24"
```

**Review by lizzcondrey — CHANGES_REQUESTED** [review]

**Review by lizzcondrey — APPROVED** [review]

File: python/pyproject.toml [hunk 2]
```diff
@@ -96,8 +96,6 @@ test = [
 ]
 
 all = [
-    "aac[runtime]",
```
Comment (lizzcondrey) [inline python/pyproject.toml]: i think you have to list dependencies here as well so that it gets included with the others?

**Review by lizzcondrey — CHANGES_REQUESTED** [review]

File: python/pyproject.toml (hunk 2 above)
Comment (bunchw) [reply python/pyproject.toml]: I don't think it's necessary to include them in both places.

Comment (lizzcondrey) [reply python/pyproject.toml]: Yeah, I just dug into the install step in the pipeline and looks like it does properly grab them, so should be good to go.

**Review by lizzcondrey — APPROVED** [review]

**Pull request merged**